### PR TITLE
Fixes #9797 - Update powershell.config.json docs

### LINKED ISSUE #9797: Update about_PowerShell_Config
### Prerequisites

- [X] **Existing Issue:** Search the existing issues for this repository. If there is an issue that fits your needs do not file a new one. Subscribe, react, or comment on that issue instead.
- [X] **Descriptive Title:** Write the title for this issue as a short synopsis. If possible, provide context. For example, "Document new `Get-Foo` cmdlet" instead of "New cmdlet."

### PowerShell Version

5.1, 7.2, 7.3, 7.4

### Summary

Most of the keys are documented but not in one place and there are some missing key. 

### Details

- There are some keys that are not documented
  - WindowsPowerShellCompatibilityModuleDenyList
  - WindowsPowerShellCompatibilityNoClobberModuleList
  - DisableImplicitWinCompat
- Clarify file location by platform and scope (allusers vs. currentuser) and order of precedence
  - CurrentUser has precedence over AllUsers
  - Group Policy has precedence over config file

### Proposed Content Type

About Topic

### Proposed Title

about_PowerShell_Config

### Related Articles

_No response_

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

Update powershell.config.json docs

- Fixes [AB#62441](https://dev.azure.com/msft-skilling/cebd7ef5-4282-448b-9701-88c8637581b7/_workitems/edit/62441)
- Fixes #9797

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://learn.microsoft.com/powershell/scripting/community/contributing/overview
[style]: https://learn.microsoft.com/powershell/scripting/community/contributing/powershell-style-guide

### CODE REVIEW

File: reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md [hunk 1]
```diff
@@ -39,358 +72,380 @@ installation.
...
 
-### ExecutionPolicy
+On Windows, settings managed by Windows Group Policy take precedence over
+settings in the configuration file. Group Policy doesn't exist on non-Windows
+platforms.
 
-> [!IMPORTANT]
-> This configuration only applies on Windows platforms.
+After Group Policy, settings defined at the **AllUsers** level take precedence
+over settings defined for the **CurrentUser** level.
```
Comment (michaeltlombardi) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md:86]: This is an interesting inversion of typical setting merging, presumably for security concerns.

File: reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md [hunk 2]
```diff
@@ -39,358 +72,380 @@ installation.
...
+- `ConsoleSessionConfiguration`
+- `ModuleLogging`
+- `ProtectedEventLogging`
+- `ScriptBlockLogging`
+- `ScriptExecution`
+- `Transcription`
+- `UpdatableHelp`
+
+The `ScriptExecution` setting is used to set the PowerShell Execution Policy.
+This take precedence over the `ExecutionPolicy` setting described above.
```
Comment (michaeltlombardi) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md:163]:
```suggestion
This takes precedence over the `ExecutionPolicy` setting described above.
```

File: reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md [hunk 3]
```diff
@@ -39,358 +72,380 @@ installation.
...
-> PowerShell variables cannot be embedded in `PSModulePath` configurations.
-> `PSModulePath` configurations on Linux and macOS are case-sensitive. A
-> `PSModulePath` configuration must use valid directory separators for the
-> platform. On macOS and Linux, this means `/`. On Windows, both `/` and `\`
-> will work.
+For descriptions of the other policy settings, see the descriptions in the
+[Common configuration settings][06] section.
 
-### ExperimentalFeatures
+On Windows, PowerShell looks for the settings in the registry. Any setting
```
Comment (michaeltlombardi) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md:180]:
```suggestion
On Windows, PowerShell looks for the settings in the registry. Any settings
```

File: reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md [hunk 4]
```diff
@@ -39,358 +72,380 @@ installation.
...
+found in the registry have precedence. Next PowerShell reads the JSON
+configuration. Any settings found under `PowerShellPolicies` as long as they
+are not found in the registry. Finally, any remaining settings found at the
```
Comment (michaeltlombardi) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md:183]: This reads like a sentence that got chopped at some point, leaving the fragment "Any settings... in the registry."

File: reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md [hunk 5]
```diff
@@ -39,358 +72,380 @@ installation.
...
+  the local machine including the default PowerShell remoting endpoints or a
+  custom endpoint having specific user role capabilities.
 
-```Schema
-"LogIdentity": "<log-identity>"
+```json
+{
+  "ConsoleSessionConfiguration": {
+    "EnableConsoleSessionConfiguration": false,
+    "ConsoleSessionConfigurationName" : []
```
Comment (michaeltlombardi) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md:233]: Is this value required when enable is set to `false`?

What's the behavior when enable is set to `true` and this value is an empty array?

File: reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md [hunk 6]
```diff
@@ -39,358 +72,380 @@ installation.
...
-> [!NOTE]
-> Setting a the log level enables all log levels above it.
+### ProtectedEventLogging
+
+This setting lets you configure Protected Event Logging. The setting contains
+two subkeys:
 
-Setting this setting to **default** will be interpreted as the default value.
-The default value is **Informational**.
+- `EnableProtectedEventLogging` - If you enable this policy setting, components
```
Comment (michaeltlombardi) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md:291]: Do we have documentation about setting this up anywhere? The certificate in the example is a name but it's not clear where this certificate is meant to be stored.

File: reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md [hunk 7]
```diff
@@ -39,358 +72,380 @@ installation.
...
 
 ### ScriptBlockLogging
 
-### ProtectedEventLogging
+This setting controls logging of all PowerShell script input. This setting
+contains two subkeys:
 
-### Transcription
+- `EnableScriptBlockLogging` - If you enable this policy setting, PowerShell
+  Core logs the processing of commands, script blocks, functions, and scripts
```
Comment (michaeltlombardi) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md:380]:
```suggestion
- `EnableScriptBlockLogging` - If you enable this policy setting, PowerShell
  logs the processing of commands, script blocks, functions, and scripts
```

File: reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md [hunk 8]
```diff
@@ -39,358 +72,380 @@ installation.
...
+PowerShell enables transcription for all PowerShell sessions.
 
-This configuration has more settings explicitly set:
+This setting controls how transcription works in PowerShell. This setting
+contains three subkeys:
 
-- Execution policy for this PowerShell installation is `AllSigned`
-- The CurrentUser module path is set to a module directory on a shared drive
-- The `PSImplicitRemotingBatching` experimental feature is enabled
+- `EnableTranscripting` - When this setting is enable, PowerShell creates
```
Comment (michaeltlombardi) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md:403]:
```suggestion
- `EnableTranscripting` - When this setting is enabled, PowerShell creates
```

File: reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md (hunk 4 above)
Comment (sdwheeler) [reply reference/7.2/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md:183]: fixed

**Review by michaeltlombardi — APPROVED** [review]

**Pull request merged**